gravity-forms/gw-format-date-merge-tags.php

```diff
@@ -9,50 +9,99 @@
+
+			$input_id = $match[1];
+			$field    = GFFormsModel::get_field( $form, $input_id );
+
+			if ( ! $field || $field->get_input_type() !== 'date' ) {
+				continue;
+			}
 
-		$format      = $field->dateFormat ? $field->dateFormat : 'mdy';
-		$parsed_date = GFCommon::parse_date( $value, $format );
+			$i        = $match[0][0] === '{' ? 4 : 5;
+			$modifier = rgar( array_map( 'trim', explode( ',', rgar( $match, $i ) ) ), 0 );
+			if ( ! $modifier ) {
+				continue;
+			}
 
-		// On the Notifications/Confirmation side, & gets encoded to &amp;. Decode it back.
-		$modifier = htmlspecialchars_decode( $modifier );
+			$value = GFFormsModel::get_lead_field_value( $entry, $field );
+			$value = $field->get_value_merge_tag( $value, $input_id, $entry, $form, $modifier, $value, $url_encode, $esc_html, $format, $nl2br );
+			if ( ! $value ) {
+				continue;
+			}
 
-		// For whatever reason, Populate Anything's LMTs works better with `&comma` than `&#44;`. But... date() doesn't
-		// like it so let's replace it before we pass it to date().
-		$modifier = str_replace( '&comma;', ',', $modifier );
+			$format      = $field->dateFormat ? $field->dateFormat : 'mdy';
+			$parsed_date = GFCommon::parse_date( $value, $format );
 
-		// phpcs:ignore WordPress.DateTime.RestrictedFunctions.date_date
-		$replace = date( $modifier, strtotime( sprintf( '%d-%d-%d', $parsed_date['year'], $parsed_date['month'], $parsed_date['day'] ) ) );
+			// On the Notifications/Confirmation side, & gets encoded to &amp;. Decode it back.
+			$modifier = htmlspecialchars_decode( $modifier );
 
-		$text = str_replace( $match[0], $replace, $text );
+			// For whatever reason, Populate Anything's LMTs works better with `&comma` than `&#44;`. But... date() doesn't
+			// like it so let's replace it before we pass it to date().
+			$modifier = str_replace( '&comma;', ',', $modifier );
 
+			$timestamp = strtotime( sprintf( '%d-%d-%d', $parsed_date['year'], $parsed_date['month'], $parsed_date['day'] ) );
+
+			if ( $locale ) {
+				$formatter = datefmt_create(
+					$locale,
+					IntlDateFormatter::FULL,
+					IntlDateFormatter::FULL,
+					null,
+					null,
+					$modifier
+				);
+				$replace   = $formatter->format( $timestamp );
+			} else {
+			// phpcs:ignore WordPress.DateTime.RestrictedFunctions.date_date
+				$replace = date( $modifier, $timestamp );
+			}
+
+			$text = str_replace( $match[0], $replace, $text );
+		}
+
+		return $text;
 	}
 
-	return $text;
-}, 10, 7 );
+	public function is_applicable_form( $form ) {
+		$form_id = isset( $form['id'] ) ? $form['id'] : $form;
+
+		return empty( $this->_args['form_id'] ) || (int) $form_id === (int) $this->_args['form_id'];
+	}
+}
+
+# Configuration
+
+new GW_Format_Date_Merge_Tag( array(
+	'form_id' => 123,
+	'locale'  => 'fr_FR', // When this option is enabled, use the ISO date format codes instead. See https://framework.zend.com/manual/1.12/en/zend.date.constants.html#zend.date.constants.selfdefinedformats
```

Comment:
I'd comment this line out, as most people likely won't want French formatted dates 😃